test/e2e/storage/testsuites/volume_expand.go

```diff
@@ -184,8 +184,18 @@ func (v *volumeExpandTestSuite) DefineTests(driver storageframework.TestDriver,
 			newSize := currentPvcSize.DeepCopy()
 			newSize.Add(resource.MustParse("1Gi"))
 			framework.Logf("currentPvcSize %v, newSize %v", currentPvcSize, newSize)
-			_, err = ExpandPVCSize(ctx, l.resource.Pvc, newSize, f.ClientSet)
-			gomega.Expect(err).To(gomega.MatchError(apierrors.IsForbidden, "While updating non-expandable PVC"))
+			gomega.Eventually(ctx, func(ctx context.Context) error {
```

Comment:
I'm curious about one thing, why does switching to `gomega.Eventually` implementation prevent this problem?

Reply:
> I'm curious about one thing, why does switching to `gomega.Eventually` implementation prevent this problem?

As I mentioned in issue #125931, the function `ExpandPVCSize` calls `c.CoreV1().PersistentVolumeClaims(origPVC.Namespace).Get`. However, if the `Get` call fails, the function `ExpandPVCSize` will immediately return an error without retrying. On the other hand, `gomega.Eventually` can perform retries, and for this particular case, I believe `gomega.Eventually` is more appropriate.

Reply:
Isn't there any retry logic here?
```go
waitErr := wait.PollImmediate(resizePollInterval, 30*time.Second, func() (bool, error) {
		var err error
		updatedPVC, err = c.CoreV1().PersistentVolumeClaims(origPVC.Namespace).Get(ctx, pvcName, metav1.GetOptions{})
		if err != nil {
			return false, fmt.Errorf("error fetching pvc %q for resizing: %w", pvcName, err)
		}

		updatedPVC.Spec.Resources.Requests[v1.ResourceStorage] = size
		updatedPVC, err = c.CoreV1().PersistentVolumeClaims(origPVC.Namespace).Update(ctx, updatedPVC, metav1.UpdateOptions{})
		if err != nil {
			framework.Logf("Error updating pvc %s: %v", pvcName, err)
			lastUpdateError = err
			return false, nil
		}
		return true, nil
	})
	if wait.Interrupted(waitErr) {
		return nil, fmt.Errorf("timed out attempting to update PVC size. last update error: %w", lastUpdateError)
	}
	if waitErr != nil {
		return nil, fmt.Errorf("failed to expand PVC size (check logs for error): %v", waitErr)
	}
```